clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ImageCachingAgent.groovy

```diff
@@ -159,6 +159,7 @@ class ImageCachingAgent implements CachingAgent, AccountAware, DriftMetric, Cust
         namedImageCacheDataMap.put(namedImageId, new DefaultCacheData(namedImageId, [
           name              : image.name,
           virtualizationType: image.virtualizationType,
+          architecture      : image.architecture,
```

Comment:
Is there anything that uses this field?  Without some change to [AmazonNamedImageLookupController](https://github.com/spinnaker/clouddriver/blob/8147e9b9244dd6fd2044737acc63545016f4a2ce/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/controllers/AmazonNamedImageLookupController.groovy), what does this accomplish?

Also, is this (NAMED_IMAGES) a good place for this information?  It already feels unsavory having virtualizationType and creationDate here, since it's possible for two different images with the same name to have different values...The same is true for architecture as well though, right?  Is it enough to have it in IMAGES, which I think 
```
Map<String, Object> attributes = objectMapper.convertValue(image, ATTRIBUTES)
```
and
```
imageCacheData.add(new DefaultCacheData(imageId, attributes, [(NAMED_IMAGES.ns): [namedImageId]]))
```
already does.

Reply:
Great callouts! @dbyron-sf 

As far as the refactoring of images and named images go, it might make sense to tackle that separately. I am not very well versed with the images API and such a refactor will probably need a deep dive into its impact. 

I verified that adding architecture to attributes is not enough and won't reflect in the API response. 

----

The architecture addition in named images, `ImageCachingAgent` is rendered by [AmazonNamedImageLookupController](https://github.com/spinnaker/clouddriver/blob/8147e9b9244dd6fd2044737acc63545016f4a2ce/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/controllers/AmazonNamedImageLookupController.groovy) as is. So, adding `architecture` here reflected in the response of the find images API, when the parameter is the name of an image (e.g. http://localhost:8084/images/find?q=hello*). 

However, a change to [AmazonNamedImageLookupController](https://github.com/spinnaker/clouddriver/blob/8147e9b9244dd6fd2044737acc63545016f4a2ce/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/controllers/AmazonNamedImageLookupController.groovy) is required for responses to find images API with image ID parameter (e.g. http://localhost:8084/images/find?q=ami-23751835) for architecture to be included in its response. Thanks for bringing my attention to this. I have updated the PR. Here is a screenshot of a sample: 
![image](https://user-images.githubusercontent.com/3614196/161819793-51e62f12-c094-42b7-a855-5230fee70489.png)

Reply:
Thanks.  https://github.com/spinnaker/clouddriver/pull/5609/commits/1d3b82efd68d9c6de1d5c84144eb308ef8ba5c8e looks right to me.